### ISSUE TITLE
Rule: Flake8-bugbear B038 " detect changes to iterable object of loop"

### ISSUE THREAD

**Issue by mimre25** [astral-sh/ruff#9511]
I've added a new rule to flake8-bugbear.

In brief, the rule detects bugs like the following:
```py
some_list = [1,2,3]
for i in some_list:
  if i % 2 == 0:
    some_list.remove(i)
  print(i)
```

(More context at github.com/PyCQA/flake8-bugbear/issues/445)

I'd be happy to contribute if its desired to implement this rule in ruff as well :slightly_smiling_face:

**Comment by charliermarsh** [comment]
Definitely happy to include in Ruff, would welcome a contribution!

**Comment by mimre25** [comment]
Sounds good - I hope I can get the PR during this week :slightly_smiling_face:

**Comment by charliermarsh** [comment]
Awesome :) I will assign you to avoid others hopping on it accidentally, but just LMK if you find yourself unable to get to it.

**Comment by mimre25** [comment]
FYI: I got a prototype working, but with flake8-bugbear we already got reports about some oversights leading to false positives - I'm looking into making it more solid - just wanted to give a heads up that I'm getting there :slightly_smiling_face:

**Comment by mimre25** [comment]
PR is out: https://github.com/astral-sh/ruff/pull/9578
